### PR TITLE
Remove Django 1.8 / 1.10 compatibility

### PR DESCRIPTION
**Pull request by gasman**

### CODE REVIEW

File: wagtail/wagtailcore/middleware.py [hunk 1]
```diff
@@ -1,15 +1,9 @@
 from __future__ import absolute_import, unicode_literals
 
-import django
+from django.utils.deprecation import MiddlewareMixin
```
Comment (gasman) [inline wagtail/wagtailcore/middleware.py]: This is a Django <1.10 compatibility shim https://docs.djangoproject.com/en/1.11/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware and should probably be eliminated too now.

File: wagtail/wagtailredirects/middleware.py [hunk 2]
```diff
@@ -1,19 +1,13 @@
 from __future__ import absolute_import, unicode_literals
 
-import django
 from django import http
+from django.utils.deprecation import MiddlewareMixin
```
Comment (gasman) [inline wagtail/wagtailredirects/middleware.py]: This is a Django <1.10 compatibility shim https://docs.djangoproject.com/en/1.11/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware and should probably be eliminated too now.

File: wagtail/wagtailcore/middleware.py (hunk 1 above)
Comment (kaedroho) [reply wagtail/wagtailcore/middleware.py]: class-based middleware is still usable in Django 1.11 though

**Comment by gasman** [comment]
Merged in 2d2518a571fe2890b3d5c6d624a73dc7a75329a2. Will add MiddlewareMixin as a separate task on #3917

File: wagtail/wagtailcore/middleware.py (hunk 1 above)
Comment (gasman) [reply wagtail/wagtailcore/middleware.py]: Yep, that's true - it should continue to work for as long as `django.utils.deprecation.MiddlewareMixin` exists, and there's no indication that Django intends to remove it any time soon. It's more of a nice-to-have, in the interests of eliminating legacy code and fully following Django >=1.10 conventions.

**Pull request closed without merging**